### PR TITLE
Added the bra and ket commands for Dirac notation support

### PR DESCRIPTION
**Pull request by jclapis**
This pull request adds the `\bra{}` and `\ket{}` commands, which are useful shorthand commands for work involving quantum mechanics and Dirac notation.

Example of a common use case:
image

Their implementation is inspired by the popular [braket LaTeX package](https://www.ctan.org/pkg/braket).

### CODE REVIEW

File: CSharpMath/Atom/LaTeXParser.cs [hunk 1]
```diff
@@ -456,6 +456,12 @@ class InnerEnvironment : IEnvironment {
           var operatorname = ReadString();
           if (!ExpectCharacter('}')) { SetError("Expected }"); return null; }
           return new LargeOperator(operatorname, null);
+        case "bra":
+          var braContents = BuildInternal(true);
+          return new Inner(new Boundary("〈"), braContents ?? new MathList(), new Boundary("|"));
```
Comment (Happypig375) [inline CSharpMath/Atom/LaTeXParser.cs:461]: When `BuildInternal` returns `null`, an parsing error has occured. `braContents` and `kerContents` should be checked for `null` and return `null`.

File: CSharpMath.CoreTests/LaTeXParserTest.cs [hunk 2]
```diff
@@ -193,6 +193,32 @@ void RunScriptTest
...
+      var list = ParseLaTeX(@"\ket{i}");
+      Assert.Collection(list,
+        CheckAtom<Inner>("", inner => {
+          Assert.Equal("|", inner.LeftBoundary.Nucleus);
+          Assert.Equal("〉", inner.RightBoundary.Nucleus);
+          Assert.Collection(inner.InnerList, CheckAtom<Variable>("i"));
+        })
+      );
+      Assert.Equal(@"\ket{i}", LaTeXParser.MathListToLaTeX(list).ToString());
+    }
```
Comment (Happypig375) [inline CSharpMath.CoreTests/LaTeXParserTest.cs:220]: Additional tests regarding parsing errors inside `\bra` and `\ket` should be added to `TestErrors` at the bottom of this file. The current implementation doesn't handle these cases correctly.

**Comment by Happypig375** [comment]
The documentation states that `\bra` and `\ket` are small versions but the current implementation corresponds to `\Bra` and `\Ket`. Please adjust accordingly.

Also, a comment should be left linking to the package provided in the post.

File: CSharpMath.CoreTests/LaTeXParserTest.cs (hunk 2 above)
Comment (jclapis) [reply CSharpMath.CoreTests/LaTeXParserTest.cs:220]: Ok, I replicated some of the parsing error tests for powers and `\sqrt`.

File: CSharpMath/Atom/LaTeXParser.cs (hunk 1 above)
Comment (jclapis) [reply CSharpMath/Atom/LaTeXParser.cs]: Fixed in the latest commit.

**Comment by jclapis** [comment]
I've made all of the requested changes. Let me know if there's anything else you need.

**Comment by Happypig375** [comment]
@jclapis Thank you for your contribution!

**Pull request merged**